### PR TITLE
Clean up elide predicate

### PR DESCRIPTION
**Pull request by clayreimann**
Depends on #270

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by xiaoyao1991** [comment]
👍

File: elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java [hunk 1]
```diff
@@ -13,191 +13,265 @@
...
-                    throw new InvalidPredicateException("LE can only take one argument");
-                }
-                Object val = PersistentResource.getValue(entity, field, dictionary);
-                return val != null && getComparisonResult(val, values.get(0)) >= 0;
-            };
+            return Operator.ge(field, values, dictionary);
         }
     };
 
-    @Getter private final String string;
```
Comment (xiaoyao1991) [inline elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java]: wow, I didn't notice we have such a bad naming exist

File: elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java [hunk 2]
```diff
@@ -13,191 +13,265 @@
...
+
+            return val != null && values.stream()
+                    .map(v -> CoerceUtil.coerce(v, val.getClass()))
+                    .anyMatch(val::equals);
+        };
+    }
+
+    //
+    // Predicate generation
+    //
```
Comment (xiaoyao1991) [inline elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java]: Maybe we should use `/**` here?

Comment (clayreimann) [inline elide-core/src/main/java/com/yahoo/elide/core/filter/Operator.java]: Not really javadoc, I just wanted some visual seperation

**Comment by xiaoyao1991** [comment]
👍  LGTM

**Comment by DeathByTape** [comment]
👍

**Pull request merged**